src/linker/Linker/PInvokeInfo.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace Mono.Linker
+{
+	[DataContract]
+	public class PInvokeInfo : IComparable
+	{
+		[DataMember (Name = "assembly")]
+		internal string AssemblyName { get; set; }
+
+		[DataMember (Name = "entryPoint")]
+		internal string EntryPoint { get; set; }
+
+		[DataMember (Name = "fullName")]
+		internal string FullName { get; set; }
+
+		[DataMember (Name = "moduleName")]
+		internal string ModuleName { get; set; }
+
+		public int CompareTo (object obj)
+		{
+			if (obj == null) return 1;
+
+			PInvokeInfo compareTo = obj as PInvokeInfo;
+			int compareField = string.Compare (this.AssemblyName, compareTo.AssemblyName);
```

Comment:
Do you specifically want culture-aware comparison? Otherwise this is much better and faster:

```suggestion
			int compareField = string.Compare (this.AssemblyName, compareTo.AssemblyName, StringComparison.Ordinal);
```

and same for the comparisons below.